diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf5f0172..b2578338 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -200,22 +200,4 @@ set(STABLEHLO_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 set(STABLEHLO_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
 set(STABLEHLO_TOOLS_DIR ${CMAKE_BINARY_DIR}/bin)
 
-# Target that provides comprehensive testing for the StableHLO repository.
-# It includes both quick and slow tests (see check-stablehlo-quick and
-# check-stablehlo-slow below).
-add_custom_target(check-stablehlo-ci)
-
-# Target that aggregates slow tests. Long-running test targets should go
-# into dependencies of this target, and we shouldn't expect humans to run them
-# regularly during development.
-add_custom_target(check-stablehlo-slow)
-add_dependencies(check-stablehlo-ci check-stablehlo-slow)
-
-# Target that aggregates fast tests. We shouldn't add long-running test targets
-# as a dependency of this target, since we'd want humans to routinely use this
-# target during development.
-add_custom_target(check-stablehlo-quick)
-add_dependencies(check-stablehlo-ci check-stablehlo-quick)
-
 add_subdirectory(stablehlo)
-add_subdirectory(examples)
diff --git a/stablehlo/CMakeLists.txt b/stablehlo/CMakeLists.txt
index 69d4196d..5828f6a5 100644
--- a/stablehlo/CMakeLists.txt
+++ b/stablehlo/CMakeLists.txt
@@ -17,7 +17,4 @@ add_subdirectory(conversions)
 add_subdirectory(dialect)
 add_subdirectory(integrations)
 add_subdirectory(reference)
-add_subdirectory(tests)
-add_subdirectory(testdata)
-add_subdirectory(tools)
 add_subdirectory(transforms)
diff --git a/stablehlo/conversions/linalg/CMakeLists.txt b/stablehlo/conversions/linalg/CMakeLists.txt
index 7b01b844..a754ec28 100644
--- a/stablehlo/conversions/linalg/CMakeLists.txt
+++ b/stablehlo/conversions/linalg/CMakeLists.txt
@@ -11,5 +11,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_subdirectory(tests)
 add_subdirectory(transforms)
diff --git a/stablehlo/conversions/tosa/CMakeLists.txt b/stablehlo/conversions/tosa/CMakeLists.txt
index d63c5baf..b718de5f 100644
--- a/stablehlo/conversions/tosa/CMakeLists.txt
+++ b/stablehlo/conversions/tosa/CMakeLists.txt
@@ -12,5 +12,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_subdirectory(tests)
 add_subdirectory(transforms)
diff --git a/stablehlo/integrations/c/CMakeLists.txt b/stablehlo/integrations/c/CMakeLists.txt
index 1efe9fd7..23909307 100644
--- a/stablehlo/integrations/c/CMakeLists.txt
+++ b/stablehlo/integrations/c/CMakeLists.txt
@@ -13,14 +13,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_mlir_public_c_api_library(CheckCAPI
-  PARTIAL_SOURCES_INTENDED
-  CheckDialect.cpp
-
-  LINK_LIBS PUBLIC
-  CheckOps
-)
-
 add_mlir_public_c_api_library(ChloCAPI
   PARTIAL_SOURCES_INTENDED
   ChloAttributes.cpp
diff --git a/stablehlo/integrations/cpp/builder/CMakeLists.txt b/stablehlo/integrations/cpp/builder/CMakeLists.txt
index 28444f4c..b84661dc 100644
--- a/stablehlo/integrations/cpp/builder/CMakeLists.txt
+++ b/stablehlo/integrations/cpp/builder/CMakeLists.txt
@@ -132,15 +132,3 @@ add_mlir_library(StablehloBuilder
   MLIRInferTypeOpInterface
   MLIRSupport  # LLVMSupport is automatically added in AddLLVM.cmake.
 )
-
-if (TARGET llvm_gtest)
-    set_target_properties(check-stablehlo-ci PROPERTIES FOLDER "Tests")
-    add_unittest(check-stablehlo-ci "unittests"
-      MlirBuilderTest.cpp
-      StablehloBuilderTest.cpp
-      AttrTypeBuilderUtilTest.cpp
-    )
-  else()
-    message(WARNING "gtest not found, unittests will not be available")
-  endif()
-
diff --git a/stablehlo/integrations/python/CMakeLists.txt b/stablehlo/integrations/python/CMakeLists.txt
index d8ef0e3e..d744f0d5 100644
--- a/stablehlo/integrations/python/CMakeLists.txt
+++ b/stablehlo/integrations/python/CMakeLists.txt
@@ -22,18 +22,6 @@ include(AddMLIRPython)
 # putting .td and .py files under . instead of mlir/python will break things,
 # even if the build rules below are adjusted accordingly.
 
-declare_mlir_python_sources(CheckPythonSources)
-declare_mlir_python_sources(CheckPythonSources.Dialects
-  ADD_TO_PARENT CheckPythonSources
-)
-
-declare_mlir_dialect_python_bindings(
-  ADD_TO_PARENT CheckPythonSources.Dialects
-  ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/mlir"
-  TD_FILE dialects/CheckOps.td
-  SOURCES dialects/check.py
-  DIALECT_NAME check)
-
 declare_mlir_python_sources(ChloPythonSources)
 declare_mlir_python_sources(ChloPythonSources.Dialects
   ADD_TO_PARENT ChloPythonSources
@@ -118,19 +106,6 @@ if(NANOBIND_WARNING_FLAGS)
   add_compile_options(${NANOBIND_WARNING_FLAGS})
 endif()
 
-declare_mlir_python_sources(CheckPythonExtensions)
-declare_mlir_python_extension(CheckPythonExtensions.Main
-  MODULE_NAME _check
-  ADD_TO_PARENT CheckPythonExtensions
-  PYTHON_BINDINGS_LIBRARY nanobind
-  SOURCES
-  CheckModule.cpp
-  EMBED_CAPI_LINK_LIBS
-  CheckCAPI
-  PRIVATE_LINK_LIBS
-    LLVMSupport
-)
-
 declare_mlir_python_sources(ChloPythonExtensions)
 declare_mlir_python_extension(ChloPythonExtensions.Main
   MODULE_NAME _chlo
@@ -189,8 +164,6 @@ add_mlir_python_common_capi_library(StablehloUnifiedPythonCAPI
   DECLARED_SOURCES
     MLIRPythonSources
     MLIRPythonExtension.RegisterEverything
-    CheckPythonSources
-    CheckPythonExtensions
     ChloPythonSources
     ChloPythonExtensions
     StablehloPythonSources
@@ -205,8 +178,6 @@ add_mlir_python_modules(StablehloUnifiedPythonModules
   DECLARED_SOURCES
     MLIRPythonSources
     MLIRPythonExtension.RegisterEverything
-    CheckPythonSources
-    CheckPythonExtensions
     ChloPythonSources
     ChloPythonExtensions
     StablehloPythonSources
@@ -217,8 +188,3 @@ add_mlir_python_modules(StablehloUnifiedPythonModules
     StablehloUnifiedPythonCAPI
 )
 
-################################################################################
-# Tests
-################################################################################
-
-add_subdirectory(tests)
